### PR TITLE
Initial attempt to add Lua to the DOSBOX debugger

### PR DESCRIPTION
**Pull request by naerbnic**
# Description

A draft PR for review to show one approach to adding a Lua interpreter to the debugger.

This does not have much functionality at the moment, but allows for an interpreter class to add functions to the lua environment.

## Related issues

_Related issues can be listed here (remove the section if not applicable.)_

# Manual testing

_Please describe the tests that you ran to verify your changes._

_Provide clear instructions so the reviewers can reproduce and verify your results._

_Include relevant details for your test configuration, operating system, etc._

_Ideally, you would also test your changes with a [sanitizer build](https://github.com/dosbox-staging/dosbox-staging/blob/main/BUILD.md#make-a-sanitizer-build) and confirm no issues were found._

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [ ] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [ ] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [ ] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/debug/debug.cpp [hunk 1]
```diff
@@ -1073,6 +1077,7 @@ bool ChangeRegister(char* str)
 }
 
 bool ParseCommand(char* str) {
+	lua_interpreter->RunCommand("print(\"Hello World\\n\")");
```
Comment (japsmits) [inline src/debug/debug.cpp:1080]: Nice! 😄 

Maybe change it to something more informational, like "Lua says Hello World\n"? Then the oblivious observer knows where this line comes from.

Comment (johnnovak) [reply src/debug/debug.cpp:1080]: I like that 😎

**Comment by johnnovak** [comment]
@naerbnic Thanks for the patience. Now is the time to finalise this, then the team will review it in the coming weeks.

**Comment by johnnovak** [comment]
@LowLevelMahn Given you seem to be experienced with the debugger, would you be interested in reviewing this change? Not just at the code level, but at the conceptual level. We're interested in your thoughts.

**Comment by LowLevelMahn** [comment]
> @LowLevelMahn Given you seem to be experienced with the debugger, would you be interested in reviewing this change? Not just the code, but at the conceptual level too. We're interested in your thoughts and insights.

there isn't much to say - there is a new "eval" command that can run Lua code - with no dependencies to the dosbox internas (so no memory access etc. possible) - so more or less no debugging related functionality so far

@naerbnic - you need to give a brief overview what features you want to provide
-eval based waching of variable content
-hocking function calls
...

still a good start for integrating Lua

**Comment by johnnovak** [comment]
> > @LowLevelMahn Given you seem to be experienced with the debugger, would you be interested in reviewing this change? Not just the code, but at the conceptual level too. We're interested in your thoughts and insights.
> 
> there isn't much to say - there is a new "eval" command that can run Lua code - with no dependencies to the dosbox internas (so no memory access etc. possible) - so more or less no debugging related functionality so far

Thanks for the feedback @LowLevelMahn 

> @naerbnic - you need to give a brief overview what features you want to provide -eval based waching of variable content -hooking function calls -print values/structures from different memory regions -access hardware/memory in real/protected mode ...

Yep, I tend to agree on these points.

> still a good start for integrating Lua - but it should not stay in this "Hello World" state - or else i would not accept the PR :)

👍🏻

**Comment by johnnovak** [comment]
Closing due to inactivity. Please feel free to raise another PR if you want to finish the work @naerbnic.

**Pull request closed without merging**